os/os.module.go

```diff
@@ -0,0 +1,241 @@
+			Name: "os",
+			Doc:  "Miscellaneous operating system interfaces",
+		},
+		Methods: methods,
+		Globals: globals,
+	})
+}
+
+// gets and parses all process enviroment variables
+func getEnvVariables() py.StringDict {
+	env_variables := os.Environ() // this gets all the enviroment variables
+	dict := py.NewStringDict()
+	for _, evar := range env_variables {
+		key_value := strings.SplitN(evar, "=", 2) // returns a []string containing [key,value]
+		dict.M__setitem__(py.String(key_value[0]), py.String(key_value[1]))
+	}
+
+	return dict
+}
+
+// gets the current working directory
+func getCwd(self py.Object, args py.Tuple) (py.Object, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return nil, py.ExceptionNewf(py.OSError, "Unable to get current working directory.")
+	}
+	return py.String(dir), nil
+}
+
+// gets the current working directory in a byte list
+func getCwdb(self py.Object, args py.Tuple) (py.Object, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return nil, py.ExceptionNewf(py.OSError, "Unable to get current working directory.")
+	}
+	return py.Bytes([]byte(dir)), nil
+}
+
+// change current working directory
+func chdir(self py.Object, args py.Tuple) (py.Object, error) {
+	if len(args) == 0 {
+		return nil, py.ExceptionNewf(py.TypeError, "Missing required argument 'path' (pos 1)")
+	}
+	if objectIsString(args[0]) {
+		dir, err := py.ReprAsString(args[0])
+		if err != nil {
+			return nil, py.ExceptionNewf(py.TypeError, "Failed to parse string") // never going to run
+		}
+		dir = strings.ReplaceAll(dir, "'", "")
+		err = os.Chdir(dir)
+		if err != nil {
+			return nil, py.ExceptionNewf(py.NotADirectoryError, "Couldn't change cwd; "+err.Error())
+		}
+		return py.None, nil
+	}
+	return nil, py.ExceptionNewf(py.TypeError, "str expected, not "+args[0].Type().Name)
+}
+
+// get a enviroment variable by key
+// Return the value of the environment variable key if it exists, or default if it doesn’t. key, default and the result are str.
+func getenv(self py.Object, args py.Tuple) (py.Object, error) {
+	var key py.Object      // key to search for
+	var default_ py.Object // return when not found
+	var err error          // error obj
+
+	if len(args) == 1 {
+		if objectIsString(args[0]) {
+			key = args[0]
+		} else {
+			return nil, py.ExceptionNewf(py.TypeError, "str expected (pos 1), not "+args[0].Type().Name)
+		}
+		default_ = py.None
+	} else if len(args) == 2 {
+		if objectIsString(args[0]) {
+			key = args[0]
+		} else {
+			return nil, py.ExceptionNewf(py.TypeError, "str expected (pos 1), not "+args[0].Type().Name)
+		}
+		if objectIsString(args[1]) {
+			key = args[0]
+			default_ = args[1]
+		} else {
+			return nil, py.ExceptionNewf(py.TypeError, "str expected (pos 2), not"+args[1].Type().Name)
+		}
+	} else {
+		return nil, py.ExceptionNewf(py.TypeError, "missing one required argument: 'name:str'")
+	}
+	var res py.Object // hold the result value
+	res, err = getEnvVariables().M__getitem__(key)
```

Comment:
creating the full `py.Dict` of environment variables, just to pick one value sounds a bit expensive...
couldn't we be a bit cleverer?
we have tested above that the provided arguments are `str`s, so we could just use the Go `os.Getenv`/`os.LookupEnv` functions.